raiden-ts/src/utils/error.ts

```diff
@@ -0,0 +1,88 @@
+import * as t from 'io-ts';
+import { map } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+
+import { PfsErrorCodes } from '../path/errors';
+
+export const ErrorDetails = t.array(
+  t.type({
+    key: t.string,
+    value: t.union([t.string, t.number]),
+  }),
+);
+export interface ErrorDetails extends t.TypeOf<typeof ErrorDetails> {}
+
+export default class RaidenError extends Error {
+  code: string;
+  details?: ErrorDetails;
+
+  constructor(message: string, details?: ErrorDetails) {
+    super(message || 'General Error');
+    this.name = 'RaidenError';
+    this.code = this.getCode(message);
+    this.details = details;
+  }
+
+  getCode(message: string): string {
+    return message ?? 'RAIDEN_ERROR';
+  }
+}
+
+export class PfsError extends RaidenError {
+  constructor(message: PfsErrorCodes, details?: ErrorDetails) {
+    super(message, details);
+    this.name = 'PfsError';
+  }
+
+  getCode(message: string): string {
```

Comment:
I don't understand the reason for a code. If the `code <=> message` mapping is already 1x1, they're redundant, and you can go with message only

Reply:
* `message` is a non-localized message for the developer for debugging/logging
* `code` is used to implement localized end-user messages

Reply:
So ideally, we are going hook up each error `code` with a locale string in our i18n setup

Reply:
Got it. I'm not a big fan of including yet another non-standard property in these error classes, but this is just personal taste, I got your intention and it's correct from this PoV. Thanks

Reply:
I also wish that `code` would be part of `Error`, never understood why it isn't 🤷‍♂